pom.xml

```diff
@@ -872,6 +872,36 @@
         <java.level>8</java.level>
       </properties>
     </profile>
+    <profile>
+      <id>deprecated-java.level-7</id>
+      <activation>
+        <property>
+          <name>java.level</name>
+          <value>7</value>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-antrun-plugin</artifactId>
+            <executions>
+              <execution>
+                <phase>initialize</phase>
+                <id>warn-deprecated-java.level</id>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <configuration>
+                  <tasks>
+                    <echo>WARNING: Plugin POM defines java.level=${java.level} which is deprecated in the current Plugin POM version. See https://github.com/jenkinsci/plugin-pom/blob/master/README.md#java-support</echo>
```

Comment:
As an aside, maybe for #133, we ought to be deprecating this property and introducing a new one which expects an actual Java specification version—thus `1.8`, `11`, etc.

Reply:
Agreed

Reply:
I agree this would properly be a good time to do it 😓

Reply:
Maybe possible to combine the tasks? Have a profile issuing a warning with two different clauses if you have `java.level` defined at all, and introduce the new property defaulting to `1.${java.level}`. Just a thought; it is not a blocker for this PR.